Slash/Utility/Comments/Comments.pm

```diff
@@ -2309,32 +2338,52 @@ sub nextPrevLinks {
 sub dispCommentNoTemplate {
 	my $args = shift;
 	my $form = getCurrentForm();
-        my $user = getCurrentUser();
-        my $constants = getCurrentStatic();
-        my $gSkin = getCurrentSkin();
+	my $user = getCurrentUser();
+	my $constants = getCurrentStatic();
+	my $gSkin = getCurrentSkin();
 	my $slashdb = getCurrentDB();
 
-	my $html_out = "<li id=\"tree_$args->{cid}\" class=\"comment\">\n";
+	my $html_out = "";
 	my $show = 0;
+	my $visible = 0;
+	my $visiblenopass = 0;
+	my $legacykids = 0;
 
 	if(defined($form->{cid}) && $form->{cid} == $args->{cid}) { $show = 1; }
-	if($user->{uid} == $args->{uid} && !$user->{is_anon}) { $show = 1; }
+	if(defined($args->{options}->{show}) && $args->{options}->{show}){ $show = 1; }
+	if($user->{uid} == $args->{uid} && !$user->{is_anon} && $user->{mode} ne 'threadtos') { $show = 1; }
 	
 	# Now shit starts getting squirrely.
-		if(!defined($args->{options}->{noCollapse}) || !$args->{options}->{noCollapse}) {
+	if(!defined($args->{options}->{noCollapse}) || !$args->{options}->{noCollapse}) {
+		if(defined($args->{points}) && $args->{points} >= $user->{threshold} && !$show && $user->{mode} eq 'threadtos') {
+                        $visiblenopass = 1;
+                }
+		if(defined($args->{points}) && $args->{points} >= $user->{highlightthresh} && !$show && $user->{mode} eq 'threadtos') {
+			$visible = 1;
+		}
 
-		if($user->{mode} ne 'flat') {
-			$html_out .= "<input id=\"commentTreeHider_$args->{cid}\" type=\"checkbox\" class=\"commentTreeHider\" autocomplete=\"off\" />\n";
+		if(!defined($args->{children}) || !$args->{children}) {
+			$legacykids = $slashdb->sqlSelect("1", "comments", "pid = $args->{cid} group by pid");
+		}
+		if($user->{mode} ne 'flat' && ($args->{children} || $legacykids)) {
+			my $checked = "";
+			if($args->{lvl} > 1 && $user->{mode} eq 'threadtos' && !$show) { $checked = "checked=\"checked\""; }
+			if($user->{mode} eq 'threadtos' && ($args->{points} >= $user->{highlightthresh} || $show)) { $checked = ""; }
+			if(defined($args->{options}->{visiblekid}) && $args->{options}->{visiblekid}) {$checked = "";}
+			$html_out .= "<input id=\"commentTreeHider_$args->{cid}\" type=\"checkbox\" class=\"commentTreeHider\" autocomplete=\"off\" $checked />\n";
 		}
 
 		$html_out .= "<input id=\"commentHider_$args->{cid}\" type=\"checkbox\" class=\"commentHider\" ";
-		if(defined($args->{points}) && $args->{points} < $user->{highlightthresh} && !$show) {
+		if(defined($args->{points}) && $user->{mode} ne "threadtos" && $args->{points} < $user->{threshold} && $args->{points} < $user->{highlightthresh} && !$show ) {
```

Comment:
Not sure I am on board with this.  Now I need to set two controls to set my level if I need to go lower.  
I think we should just dump thresh for non-tos and just show Breakthrough in the options.

What do you think.